src/canopy_alloc.F90

```diff
@@ -67,22 +84,37 @@ SUBROUTINE canopy_alloc
     if(.not.allocated(ppfd_sun240))        allocate(ppfd_sun240(nlat*nlon,modlays))
     if(.not.allocated(ppfd_shade240))      allocate(ppfd_shade240(nlat*nlon,modlays))
 
-    if(.not.allocated(tmp2mref_tmp))       allocate(tmp2mref_tmp(ntime,nlat*nlon))
-    if(.not.allocated(ubzref_tmp))         allocate(ubzref_tmp(ntime,nlat*nlon))
     if(.not.allocated(daily_maxt2m))       allocate(daily_maxt2m(nlat*nlon))
     if(.not.allocated(daily_mint2m))       allocate(daily_mint2m(nlat*nlon))
     if(.not.allocated(daily_maxws10m))     allocate(daily_maxws10m(nlat*nlon))
 
-    if(.not.allocated(tleaf_sun24_tmp_3d))    allocate(tleaf_sun24_tmp_3d(ntime,nlon,nlat,modlays))
-    if(.not.allocated(tleaf_shade24_tmp_3d))  allocate(tleaf_shade24_tmp_3d(ntime,nlon,nlat,modlays))
-    if(.not.allocated(tleaf_ave24_tmp_3d))    allocate(tleaf_ave24_tmp_3d(ntime,nlon,nlat,modlays))
-    if(.not.allocated(ppfd_sun24_tmp_3d))     allocate(ppfd_sun24_tmp_3d(ntime,nlon,nlat,modlays))
-    if(.not.allocated(ppfd_shade24_tmp_3d))   allocate(ppfd_shade24_tmp_3d(ntime,nlon,nlat,modlays))
-    if(.not.allocated(tleaf_sun240_tmp_3d))   allocate(tleaf_sun240_tmp_3d(ntime,nlon,nlat,modlays))
-    if(.not.allocated(tleaf_shade240_tmp_3d)) allocate(tleaf_shade240_tmp_3d(ntime,nlon,nlat,modlays))
-    if(.not.allocated(tleaf_ave240_tmp_3d))   allocate(tleaf_ave240_tmp_3d(ntime,nlon,nlat,modlays))
-    if(.not.allocated(ppfd_sun240_tmp_3d))    allocate(ppfd_sun240_tmp_3d(ntime,nlon,nlat,modlays))
-    if(.not.allocated(ppfd_shade240_tmp_3d))  allocate(ppfd_shade240_tmp_3d(ntime,nlon,nlat,modlays))
+    if (hist_opt .eq. 1) then
+        if(.not.allocated(tleaf_sun24_tmp_3d))    allocate(tleaf_sun24_tmp_3d(25,nlon,nlat,modlays))
+        if(.not.allocated(tleaf_shade24_tmp_3d))  allocate(tleaf_shade24_tmp_3d(25,nlon,nlat,modlays))
+        if(.not.allocated(tleaf_ave24_tmp_3d))    allocate(tleaf_ave24_tmp_3d(25,nlon,nlat,modlays))
+        if(.not.allocated(ppfd_sun24_tmp_3d))     allocate(ppfd_sun24_tmp_3d(25,nlon,nlat,modlays))
+        if(.not.allocated(ppfd_shade24_tmp_3d))   allocate(ppfd_shade24_tmp_3d(25,nlon,nlat,modlays))
+        if(.not.allocated(tleaf_sun240_tmp_3d))   allocate(tleaf_sun240_tmp_3d(241,nlon,nlat,modlays))
+        if(.not.allocated(tleaf_shade240_tmp_3d)) allocate(tleaf_shade240_tmp_3d(241,nlon,nlat,modlays))
+        if(.not.allocated(tleaf_ave240_tmp_3d))   allocate(tleaf_ave240_tmp_3d(241,nlon,nlat,modlays))
+        if(.not.allocated(ppfd_sun240_tmp_3d))    allocate(ppfd_sun240_tmp_3d(241,nlon,nlat,modlays))
+        if(.not.allocated(ppfd_shade240_tmp_3d))  allocate(ppfd_shade240_tmp_3d(241,nlon,nlat,modlays))
+        if(.not.allocated(tmp2mref_tmp_3d))       allocate(tmp2mref_tmp_3d(25,nlon,nlat))
+        if(.not.allocated(ubzref_tmp_3d))         allocate(ubzref_tmp_3d(25,nlon,nlat))
+        if(.not.allocated(tleaf_sun24_rst_3d))    allocate(tleaf_sun24_rst_3d(24,nlon,nlat,modlays))
```

Comment:
Not sure we need the "rst" variables, please see my general comment on idea to output the total 24-hr and 240-hr leaf temp, ppfd in i,j,k (and maybe lai) as restart variables to be read in separately and adjusted to instantaneous based on fractions until new 24-hr and 240-hr become available.